presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1315,7 +1315,11 @@ private PhysicalOperation visitScanFilterAndProject(
             catch (PrestoException e) {
                 throw e;
             }
-            catch (RuntimeException e) {
+            catch (Exception e) {
+                if (e.getCause() instanceof  PrestoException) {
```

Comment:
If you unwrap looking for a specific exception you should keep going until you find it and not just one level.

I think maybe you want to wrap in a new PrestoException with the contents of the wrapped PrestoException.

Reply:
I was hoping none wraps PrestoException in another PrestoException.

Reply:
> I was hoping none wraps PrestoException in another PrestoException. Also I'm just mimicking the previous catch PrestoException. I have seen that pattern a few places - if you get PrestoException you just throw it again.

Reply:
So I reverted back to the old code as the flow doesn't seem to be relevant for user queries. We still block it but the message is different.